### DISCUSSION TITLE
The REEM: Red Eclipse Environment Mixer

### DISCUSSION THREAD

**Discussion by cloudziq** [redeclipse/base#1273] [Maps and Content Showcase]
Small tool which works best in 2.0.1 as is integrated into `tool mode` (under ESC key in editmode). 
To bring up mixer in 2.0, you'll need to type `/reem` while in editmode

Allows randomizing your env variables using a simple UI menu.
Good for quick creating nice-looking environments, much faster than doing the same manually, often need only 4 or 5 vars to tweak manually after use of this tool - but hey, it's still way faster than setting up every vars manually with fitting colors etc

![20210611203326](https://user-images.githubusercontent.com/13541407/121779354-8237da80-cb9b-11eb-8464-6b8051462a48.png)

[REEM DOWNLOAD](https://github.com/redeclipse/base/files/6996956/REEM.zip)

.
.

USAGE:
- unpack and place `script` folder (with REEM.cfg inside) to your RE user data folder
- add this line to autoexec.cfg if you wish this script to be included all time:
`exec script/REEM.cfg`
or use command in-game:
`/exec script/REEM.cfg`

- There are 4 main windows, `ACTIVE 1 & 2` are the main ones, a script can randomize only when one of those is selected.
  After opening it for the first time, the current environment will be preserved in the `SAVED` window - just in case you wish to restore it, or to compare it with actual results
  (warning: script can't detect if you load another map, if you do this, open REEM and hit `REPLACE SAVED`, because there is still save from the previous map)

- `GENERATE` button randomizes the environment, below you can find checkboxes if you wish to limit generation only to certain things.
  When all three RGB Sliders are set to 0, random color will be chosen
  The fourth slider defines `color consistency` across different variables (in percentage), lowering it may result in non-fitting colors (for example, the result may be like that:  green fog, blue sun, purple clouds, yellow shadows, etc)

- `MIX` button is used to mix both ACTIVE1 with ACTIVE2 environments
  There is no way to mix textures. Cloudlayer, envlayer and skybox will be used from the currently selected window, while everything else will be mixed

- When the current window is ACTIVE1 or 2, you can right-click on any other window to copy everything

- The `Red Eclipse` button just places atmo disk (sun, moon etc) at the center of the screen

- Script changes only water1 vars, all other material volumes are untouched

_**I can't get tooltips to appear, not sure how to set them up, so for now - no tooltips under the cursor, sorry**_

.
.

here are some examples where I used REEM to automatically create a similar-looking environments as these used on map:

`original:`
![20210612151755](https://user-images.githubusercontent.com/13541407/121782251-ead98400-cba8-11eb-827e-45b8c9099c01.png)

`from REEM:`
![20210612151811](https://user-images.githubusercontent.com/13541407/121782275-178d9b80-cba9-11eb-95a2-ab9135f689c4.png)

.
.

`original:`
![20210612154547](https://user-images.githubusercontent.com/13541407/121782303-2d9b5c00-cba9-11eb-8712-d0fe08d5a22d.png)

`from REEM:`
![20210612154601](https://user-images.githubusercontent.com/13541407/121782328-49066700-cba9-11eb-8549-46a390735af7.png)

As you can see, results are quite solid (more or less)
Have fun ^.^

**Comment by Sondro** [comment]
This app looks pretty cool.

I think for it to be long-term useful it would help to post the source.

**Reply by dc-redeclipse** [reply]
These functions are now integrated simular in the new editui in the upcoming next release. You can check that out already in the beta / devel version.
